rules/integrations/okta/credential_access_okta_authentication_for_multiple_users_from_single_source.toml

```diff
@@ -0,0 +1,125 @@
+- Users may share an endpoint related to work or personal use in which separate Okta accounts are used.
+    - Architecturally, this shared endpoint may leverage a proxy for security or privacy reasons.
+    - Shared systems such as Kiosks and conference room computers may be used by multiple users.
+    - Shared working spaces may have a single endpoint that is used by multiple users.
+
+### Response and remediation:
+- Review the profile of the users involved in this action to determine if proxy usage may be expected.
+- If the user is legitimate and the authentication behavior is not suspicious based on device analysis, no action is required.
+- If the user is legitimate but the authentication behavior is suspicious, consider resetting passwords for the users involves and enabling multi-factor authentication (MFA).
+    - If MFA is already enabled, consider resetting MFA for the users.
+- If any of the users are not legitimate, consider deactivating the user's account.
+- Conduct a review of Okta policies and ensure they are in accordance with security best practices.
+- Check with internal IT teams to determine if the accounts involved recently had MFA reset at the request of the user.
+    - If so, confirm with the user this was a legitimate request.
+    - If so and this was not a legitimate request, consider deactivating the user's account temporarily.
+        - Reset passwords and reset MFA for the user.
+- If this is a false positive, consider adding the `okta.debug_context.debug_data.dt_hash` field to the `exceptions` list in the rule.
+    - This will prevent future occurrences of this event for this device from triggering the rule.
+    - Alternatively adding `okta.client.ip` or a CIDR range to the `exceptions` list can prevent future occurrences of this event from triggering the rule.
+        - This should be done with caution as it may prevent legitimate alerts from being generated.
+"""
+references = [
+    "https://support.okta.com/help/s/article/How-does-the-Device-Token-work?language=en_US",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+    "https://www.elastic.co/security-labs/testing-okta-visibility-and-detection-dorothy",
+    "https://sec.okta.com/articles/2023/08/cross-tenant-impersonation-prevention-and-detection",
+    "https://www.okta.com/resources/whitepaper-how-adaptive-mfa-can-help-in-mitigating-brute-force-attacks/"
+]
+risk_score = 21
+rule_id = "94e734c0-2cda-11ef-84e1-f661ea17fbce"
+setup = "The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."
+severity = "low"
+tags = ["Use Case: Identity and Access Audit", "Data Source: Okta", "Tactic: Credential Access"]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+FROM logs-okta*
+| WHERE
+    event.dataset == "okta.system"
+    AND (event.action RLIKE "user\\.authentication(.*)" OR event.action == "user.session.start")
```

Comment:
I don't know whether this would increase performance, but I could imagine that adding the event.action without the wildcard first in the query potentially could.

Reply:
Doesn't hurt to try and adjust. Thanks for the suggestion!